src/System.Net.Security/tests/FunctionalTests/SslStreamAlertsTest.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.IO;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    public class SslStreamAlertsTest
+    {
+        private const uint SEC_E_ILLEGAL_MESSAGE = 0x80090326;
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public async Task SslStream_StreamToStream_HandshakeAlert_Ok()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new SslStream(clientStream, true, AllowAnyServerCertificate))
+            using (var server = new SslStream(serverStream, true, FailClientCertificate))
+            using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+            {
+                Task serverAuth = server.AuthenticateAsServerAsync(certificate);
+                await client.AuthenticateAsClientAsync(certificate.GetNameInfo(X509NameType.SimpleName, false));
+
+                byte[] buffer = new byte[1024];
+
+                // Schannel semantics require that Decrypt is called to receive an alert.
+                await client.WriteAsync(buffer, 0, buffer.Length);
+                var exception = await Assert.ThrowsAsync<IOException>(() => client.ReadAsync(buffer, 0, buffer.Length));
+
+                Assert.IsType<Win32Exception>(exception.InnerException);
+                var win32ex = (Win32Exception)exception.InnerException;
+
+                // The Schannel HResults for each alert are documented here: 
+                // https://msdn.microsoft.com/en-us/library/windows/desktop/dd721886(v=vs.85).aspx
+                Assert.Equal(SEC_E_ILLEGAL_MESSAGE, (uint)win32ex.NativeErrorCode);
```

Comment:
If Win32Exception and these HRESULT values are part of the contract please note that in the pending issue for the Unix implementation